crates/publish/src/oci/client.rs

```diff
@@ -209,28 +214,89 @@ impl Client {
         ))
     }
 
+    /// Save a credential set containing the registry username and password.
+    pub async fn login(
+        server: impl AsRef<str>,
+        username: impl AsRef<str>,
+        password: impl AsRef<str>,
+    ) -> Result<()> {
+        // We want to allow a user to login to both https://ghcr.io and ghcr.io.
+        let server = match server.as_ref().parse::<Url>() {
+            Ok(url) => url
+                .host_str()
```

Comment:
This is a little odd as it will strip paths too. Spec aside, that might be surprising for some people: https://github.com/distribution/distribution/issues/29

Reply:
Maybe I'm not following code flow closely enough, but this seems particularly odd given https://github.com/fermyon/spin/pull/1150/files#diff-d0a4fbcfd5b61ee9445cd726d5a0f80132f470ae85c56fdbdc4dfed7e3f48d15R372

Reply:
Those were 2 different code paths, but the extra mapping for Docker Hub is not needed anymore.

That being said, removing the path is... not ideal. Most registries can redirect, and I can't find a registry service with a path. If we do, and this becomes an issue, we would update this.